Gopkg.toml

```diff
@@ -295,7 +295,8 @@ required = [
 
 [[constraint]]
   name = "github.com/miekg/dns"
-  revision = "v1.1.4"
+  revision = "164b22ef9acc6ebfaef7169ab51caaef67390823"
+  source = "github.com/cilium/dns"
```

Comment:
I proposed this in [an upstream PR](https://github.com/miekg/dns/pull/918,) but they countered with a less invasive approach. The good thing is that the need for upstream changes was acknowledged. It may take some more time to finalize, but IMO it is safe to merge with what we have now, and then update back to the upstream branch when that is ready.